core/src/main/java/com/linecorp/armeria/internal/SslContextUtil.java

```diff
@@ -0,0 +1,405 @@
+
+import javax.net.ssl.SSLException;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.Flags;
+
+import io.netty.handler.codec.http2.Http2SecurityUtil;
+import io.netty.handler.ssl.ApplicationProtocolConfig;
+import io.netty.handler.ssl.ApplicationProtocolConfig.Protocol;
+import io.netty.handler.ssl.ApplicationProtocolConfig.SelectedListenerFailureBehavior;
+import io.netty.handler.ssl.ApplicationProtocolConfig.SelectorFailureBehavior;
+import io.netty.handler.ssl.ApplicationProtocolNames;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslProvider;
+import io.netty.handler.ssl.SupportedCipherSuiteFilter;
+
+/**
+ * Utilities for configuring {@link SslContextBuilder}.
+ */
+public final class SslContextUtil {
+
+    private static final ApplicationProtocolConfig ALPN_CONFIG = new ApplicationProtocolConfig(
+            Protocol.ALPN,
+            // NO_ADVERTISE is currently the only mode supported by both OpenSsl and JDK providers.
+            SelectorFailureBehavior.NO_ADVERTISE,
+            // ACCEPT is currently the only mode supported by both OpenSsl and JDK providers.
+            SelectedListenerFailureBehavior.ACCEPT,
+            ApplicationProtocolNames.HTTP_2,
+            ApplicationProtocolNames.HTTP_1_1);
+
+    // OpenSSL's default enabled TLSv1.3 ciphers as documented at https://wiki.openssl.org/index.php/TLS1.3
+    private static final List<String> TLS_V13_CIPHERS = ImmutableList.of("TLS_AES_256_GCM_SHA384",
+                                                                         "TLS_CHACHA20_POLY1305_SHA256",
+                                                                         "TLS_AES_128_GCM_SHA256");
+
+    public static final List<String> DEFAULT_CIPHERS = ImmutableList.<String>builder()
+            .addAll(TLS_V13_CIPHERS)
+            .addAll(Http2SecurityUtil.CIPHERS)
+            .build();
+
+    public static final List<String> DEFAULT_PROTOCOLS = ImmutableList.of("TLSv1.3", "TLSv1.2");
+
+    /**
+     * Creates a {@link SslContext} with Armeria's defaults, enabling support for HTTP/2, TLSv1.3, and
+     * TLSv1.2.
+     */
+    public static SslContext createSslContext(Supplier<SslContextBuilder> sslContextSupplier,
+                                              boolean forceHttp1,
+                                              Consumer<? super SslContextBuilder> userCustomizer) {
+        final SslContextBuilder builder = sslContextSupplier.get();
+
+        builder.sslProvider(Flags.useOpenSsl() ? SslProvider.OPENSSL : SslProvider.JDK)
+                  .protocols(DEFAULT_PROTOCOLS.toArray(new String[0]));
+
+        builder.ciphers(DEFAULT_CIPHERS, SupportedCipherSuiteFilter.INSTANCE);
+
+        userCustomizer.accept(builder);
+
+        // We call user customization logic before setting ALPN to make sure they don't break
```

Comment:
Seems better to move this comment the above? (L81)

Reply:
I think this location is clearer since it explains why we're setting ALPN here, the customizer call actually isn't the main point here. But changed the verb to `called` in case it makes it more natural.